src/vulkan-renderer/application.cpp

```diff
@@ -333,7 +341,59 @@ Application::Application(int argc, char **argv) {
             auto debug_report_ci = wrapper::make_info<VkDebugReportCallbackCreateInfoEXT>();
             debug_report_ci.flags = VK_DEBUG_REPORT_ERROR_BIT_EXT | VK_DEBUG_REPORT_WARNING_BIT_EXT |
                                     VK_DEBUG_REPORT_PERFORMANCE_WARNING_BIT_EXT | VK_DEBUG_REPORT_ERROR_BIT_EXT;
-            debug_report_ci.pfnCallback = (PFN_vkDebugReportCallbackEXT)&vulkan_debug_message_callback;
+
+            // To avoid complex dependencies to other modules, we use a lambda here to report the validation layer
+            // message. We are using two different versions of the lambda because we can't pass
+            // stop_on_validation_message to the lambda directly.
+            if (stop_on_validation_message) {
```

Comment:
A lot of duplicated code, you can merge both as the only difference is the stopping.
Which you can put inside, as passing the bool.

Reply:
That's not possible sadly because the lambda must be created on the stack. (See Discord discussion)

Reply:
Pity.

Reply:
Another option would be to make it either global or to move it into another .cpp file, but then we would need a global variable to signal that the option `--stop-on-validation-message` is specified. So this is our best option afaik.